### PR TITLE
Including Azure ML compute with existing examples. Adding network-isolated examples.

### PR DESCRIPTION
**Pull request by denniseik**
With this PR we bring multiple changes to the Azure Machine Learning examples:

- We add compute and bug fixes to the 101 example
- We introduce compute to the 201 examples
- We add a brownfield 202 example

### CODE REVIEW

**Comment by denniseik** [comment]
@grayzu @dapolloxp @ryhud @djr1991

**Comment by denniseik** [comment]
@grayzu Ping on this one, as we'd like to get our docs published after merging these pieces

File: quickstart/101-machine-learning/readme.md
Comment (grayzu) [inline quickstart/101-machine-learning/readme.md:36]: Don't we need to have the environment and location variables defined here as well?

Comment (grayzu) [reply quickstart/101-machine-learning/readme.md:36]: Actually, it looks like there are defaults defined for this so not needed here. The defaults should be shown in the readme. noted above.

File: quickstart/101-machine-learning/readme.md
Comment (grayzu) [inline quickstart/101-machine-learning/readme.md:24]: Please show the default values for environment and location in the descriptions below.

File: quickstart/201-machine-learning-moderately-secure/readme.md
Comment (grayzu) [inline quickstart/201-machine-learning-moderately-secure/readme.md]: Show the default values int he descriptions for variables with defaults.

File: quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md
Comment (grayzu) [inline quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md]: per previous suggestions, please add default values to these descriptions.

File: quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md
Comment (grayzu) [inline quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md]: All of the *_resource_id variables should be handled differently so users will not need to pass in a bunch of big convoluted strings. Can the variables take names for these instead and use [data sources](https://www.terraform.io/docs/language/data-sources/index.html) to get the required id in the resource itself? LMK if you need examples, I am happy to help.

File: quickstart/202-machine-learning-moderately-secure-existing-VNet/variables.tf
Comment (grayzu) [inline quickstart/202-machine-learning-moderately-secure-existing-VNet/variables.tf:12]: There should be a default value here to be consistent with the other examples. I believe they used "East US" as a default.

**Review by grayzu — CHANGES_REQUESTED** [review]
Overall these changes look fantastic! Thank you for the contribution!!

In addition to the individual comments that I have made inline, please run the following command from each of your quickstart directories to format the *.tf files correctly / consistently:

```bash
terraform fmt
```

**Comment by denniseik** [comment]
Thanks @grayzu ! Tagging @ryhud @dapolloxp as well to look a the feedback.

File: quickstart/101-machine-learning/readme.md
Comment (denniseik) [reply quickstart/101-machine-learning/readme.md:36]: Included some default values in the variable table.

File: quickstart/101-machine-learning/readme.md
Comment (denniseik) [reply quickstart/101-machine-learning/readme.md:24]: Yup, will include it

File: quickstart/202-machine-learning-moderately-secure-existing-VNet/variables.tf
Comment (denniseik) [reply quickstart/202-machine-learning-moderately-secure-existing-VNet/variables.tf:12]: good spot!

File: quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md
Comment (denniseik) [reply quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md]: @grayzu Are there any lean ways of doing this code-wise, to allow for flexibility to a situation where e.g. private DNS zones may reside in a different subscription opposed to the rest of the resources? Its seeking a bit of a balance between user-friendliness and comprehensiveness of the example.

Comment (denniseik) [reply quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md]: We pushed some changes to use the data construct for existing networks and subnets. Where we didn't use the data construct is private link DNS zones. Since they may come from the same subscription, another subscription or multiple subscriptions and/or resource groups, we think it would make the example a bit complicated.

File: quickstart/202-machine-learning-moderately-secure-existing-VNet/compute.tf
Comment (grayzu) [inline quickstart/202-machine-learning-moderately-secure-existing-VNet/compute.tf]: This does not need to be in the string format. it can just be
```terraform
subnet_resource_id = data.azurerm_subnet.training.id
```

This applies to `subnet_resource_id` as well as other `data.azurerm_*` references below.

**Review by grayzu — COMMENTED** [review]
minor syntax comment that is not required but good to align with the latest syntax.

File: quickstart/202-machine-learning-moderately-secure-existing-VNet/compute.tf
Comment (grayzu) [reply quickstart/202-machine-learning-moderately-secure-existing-VNet/compute.tf]: looks like this was resolved in a later merge so no longer applicable.

**Comment by grayzu** [comment]
@denniseik I am going to go ahead and merge this as the code looks good. I would suggest updating the `quickstart/202-machine-learning-moderately-secure-existing-VNet/readme.md` to show all of the required variables but will leave that up to you.

**Comment by denniseik** [comment]
Thanks @grayzu !

**Pull request merged**
